pkg/scheduler/fair_sharing_iterator.go

```diff
@@ -210,10 +210,9 @@ func (ec *entryComparer) computeDRS(rootCohort *cache.CohortSnapshot, cqToEntry
 		// calculate DRS, with workload, for all Cohorts on
 		// path to root.
 		cohort := cq.Parent()
-		for cohort.HasParent() {
-			dominantResourceShare := cohort.DominantResourceShare()
-			ec.drsValues[drsKey{parentCohort: cohort.Parent().GetName(), workloadKey: workload.Key(entry.Obj)}] = dominantResourceShare
-			cohort = cohort.Parent()
+		for c := range cohort.PathToRoot() {
+			dominantResourceShare := c.DominantResourceShare()
+			ec.drsValues[drsKey{parentCohort: c.Parent().GetName(), workloadKey: workload.Key(entry.Obj)}] = dominantResourceShare
 		}
```

Comment:
to reduce diff, and can delete cohort variable above
```suggestion
		for cohort := range cq.Parent().PathToRoot() {
			dominantResourceShare := cohort.DominantResourceShare()
			ec.drsValues[drsKey{parentCohort: cohort.Parent().GetName(), workloadKey: workload.Key(entry.Obj)}] = dominantResourceShare
		}
```